github/client.go

```diff
@@ -420,50 +420,79 @@ func (client *Client) ForkRepository(project *Project) (repo *octokit.Repository
 	return
 }
 
-func (client *Client) Issues(project *Project) (issues []octokit.Issue, err error) {
-	url, err := octokit.RepoIssuesURL.Expand(octokit.M{"owner": project.Owner, "repo": project.Name})
+type IssueParams struct {
+	Title  string   `json:"title"`
+	Body   string   `json:"body"`
+	Labels []string `json:"labels"`
+}
+
+type Issue struct {
+	Number      int          `json:"number"`
+	State       string       `json:"state"`
+	Title       string       `json:"title"`
+	Body        string       `json:"body"`
+	User        *User        `json:"user"`
+	Assignee    *User        `json:"assignee"`
+	Labels      []IssueLabel `json:"labels"`
+	PullRequest *PullRequest `json:"pull_request"`
+	HtmlUrl     string       `json:"html_url"`
+}
+
+type IssueLabel struct {
+	Name  string `json:"name"`
+	Color string `json:"color"`
+}
+
+type User struct {
+	Login string `json:"login"`
+}
+
+type PullRequest struct {
+	ApiUrl  string `json:"url"`
+	HtmlUrl string `json:"html_url"`
+}
+
+func (client *Client) FetchIssues(project *Project, filterParams map[string]interface{}) (issues []Issue, err error) {
+	api, err := client.simpleApi()
 	if err != nil {
 		return
 	}
 
-	api, err := client.api()
-	if err != nil {
-		err = FormatError("getting issues", err)
-		return
+	path := fmt.Sprintf("repos/%s/%s/issues", project.Owner, project.Name)
+	if filterParams != nil {
+		query := url.Values{}
+		for key, value := range filterParams {
+			switch v := value.(type) {
+			case string:
+				query.Add(key, v)
+			}
+		}
+		path += "?" + query.Encode()
```

Comment:
It's a minor optimization but I guess that if filterParams contains only one value that is not a string, then query.Encode() would be empty.